Fabric.Authorization.Domain/Stores/Services/PermissionService.cs

```diff
@@ -148,21 +148,28 @@ public PermissionService(IPermissionStore permissionStore, RoleService roleServi
             IEnumerable<Permission> existingAllowPermissions,
             IEnumerable<Permission> existingDenyPermissions)
         {
-            var alreadyExistAllow = allowPermissionsToAdd.Where(p => existingAllowPermissions.Contains(p));
-            var alreadyExistDeny = denyPermissionsToAdd.Where(p => existingDenyPermissions.Contains(p));
+            var invalidPermissions = new List<KeyValuePair<string,string>>();
 
-            if (alreadyExistAllow.Any() || alreadyExistDeny.Any())
+            invalidPermissions.AddRange(allowPermissionsToAdd.Intersect(existingAllowPermissions)
+                .Select(p => new KeyValuePair<string,string>("Duplicate allow permissions", p.ToString())));
+            invalidPermissions.AddRange(denyPermissionsToAdd.Intersect(existingDenyPermissions)
+                .Select(p => new KeyValuePair<string, string>("Duplicate deny permissions", p.ToString() )));
+
+            invalidPermissions.AddRange(allowPermissionsToAdd.Intersect(existingDenyPermissions)
+                .Select(p => new KeyValuePair<string, string>("Exists as deny", p.ToString() )));
+            invalidPermissions.AddRange(denyPermissionsToAdd.Intersect(existingAllowPermissions)
+                .Select(p => new KeyValuePair<string, string>("Exists as allow", p.ToString())));
```

Comment:
Do we also need to check if any of the new deny permissions overlap with the new allow permissions?

Reply:
do you mean if in a single request they add the same permission as both allow and deny?